libblkid/src/superblocks/bitlocker.c

```diff
@@ -149,26 +182,50 @@ static int probe_bitlocker(blkid_probe pr,
 {
 	const unsigned char *buf_fve = NULL;
 	const unsigned char *buf_hdr = NULL;
+	const struct bde_fve_metadata_entry *entry;
 	int rc, kind;
+	uint64_t off;
 
 	rc = get_bitlocker_headers(pr, &kind, &buf_hdr, &buf_fve);
 	if (rc)
 		return rc;
 
-	if (kind == BDE_VERSION_WIN7) {
-		const struct bde_header_win7 *hdr = (const struct bde_header_win7 *) buf_hdr;
-
-		/* Unfortunately, it seems volume_serial is always zero */
-		blkid_probe_sprintf_uuid(pr,
-				(const unsigned char *) &hdr->volume_serial,
-				sizeof(hdr->volume_serial),
-				"%016d", le32_to_cpu(hdr->volume_serial));
-	}
-
 	if (buf_fve) {
 		const struct bde_fve_metadata *fve = (const struct bde_fve_metadata *) buf_fve;
 
-		blkid_probe_sprintf_version(pr, "%d", fve->version);
+		blkid_probe_sprintf_version(pr, "%d", le16_to_cpu(fve->block_header.version));
+
+		for (off = sizeof(struct bde_fve_metadata_header);
+		     off + sizeof(struct bde_fve_metadata_entry) < le32_to_cpu(fve->header.size);
+		     off += le16_to_cpu(entry->size)) {
+			entry = (const struct bde_fve_metadata_entry *) ((const char *) &fve->header + off);
+			if (off % 2 ||
+			    le16_to_cpu(entry->size) < sizeof(struct bde_fve_metadata_entry) ||
+			    off + le16_to_cpu(entry->size) >= le32_to_cpu(fve->header.size))
+				return -1;
+
+			if (entry->entry_type == BDE_METADATA_ENTRY_TYPE_DESCRIPTION &&
```

Comment:
Here the `cpu_to_le16()` is part of `BDE_METADATA_ENTRY_TYPE_DESCRIPTION`.
Doing the endianess conversion as part of  `le16_to_cpu(entry->entry_type)` seems more uniform with the rest of the code.